Index: texlive-bin/texk/detex/detex-src/detex.man
===================================================================
--- texlive-bin.orig/texk/detex/detex-src/detex.man	2020-09-19 23:23:56.000000000 +0200
+++ texlive-bin/texk/detex/detex-src/detex.man	2020-09-21 23:57:19.434430760 +0200
@@ -73,7 +73,7 @@
 .PP
 The
 .B \-r
-option tries to naively replace $..$, $$..$$, \(..\) and \[..\]
+option tries to naively replace $..$, $$..$$, (..) and [..]
 with nouns and verbs (in particular, "noun" and "verbs")
 in a way that keeps sentences readable.
 .PP
