Date: Fri, 14 Aug 2015 16:11:50 +0200
Subject: consider Hurd as a POSIX system
Forwarded: https://github.com/LuaJIT/LuaJIT/issues/1384
Author: Enrico Tassi <gareuselesinge@debian.org>
Last-Update: 2025-08-18

---
 src/Makefile  | 3 +++
 src/lj_arch.h | 2 ++
 2 files changed, 5 insertions(+)

--- a/libs/luajit/LuaJIT-src/src/Makefile
+++ b/libs/luajit/LuaJIT-src/src/Makefile
@@ -354,6 +354,9 @@ else
   ifeq (GNU/kFreeBSD,$(TARGET_SYS))
     TARGET_XLIBS+= -ldl
   endif
+  ifeq (GNU,$(TARGET_SYS))
+    TARGET_XLIBS+= -ldl
+  endif
 endif
 endif
 endif
--- a/libs/luajit/LuaJIT-src/src/lj_arch.h
+++ b/libs/luajit/LuaJIT-src/src/lj_arch.h
@@ -96,6 +96,8 @@
 #elif defined(__QNX__)
 #define LJ_TARGET_QNX		1
 #define LUAJIT_OS	LUAJIT_OS_POSIX
+#elif defined(__GNU__)
+#define LUAJIT_OS	LUAJIT_OS_POSIX
 #else
 #define LUAJIT_OS	LUAJIT_OS_OTHER
 #endif
