--- a/utils/m-tx/mtx-src/preamble.c        2018-01-09 14:14:43.000000000 +0000
+++ b/utils/m-tx/mtx-src/preamble.c        2018-04-08 12:23:37.000000000 +0100
@@ -801,8 +801,8 @@
     stave_size[i] = unspec;
   for (i = 0; i <= maxstaves; i++)
     nspace[i] = unspec;
-  nspace[i] = unspec;
-  stave_size[i-1] = unspec;
+  /* next line seems to be spurious.  0.63a RDT */
+  /* begin  nspace[i]:=unspec;  stave_size[i]:=unspec;  end; */
   n_pages = 1;
   n_systems = 1;
   readStyles();
--- a/utils/m-tx/mtx-src/preamble.pas      2018-01-09 14:14:43.000000000 +0000
+++ b/utils/m-tx/mtx-src/preamble.pas      2018-04-08 12:21:23.000000000 +0100
@@ -501,7 +501,8 @@
   for i:=1 to maxvoices do setVocal(i,false);
   for i:=1 to maxstaves do stave_size[i]:=unspec;
   for i:=0 to maxstaves do nspace[i]:=unspec;
-  begin  nspace[i]:=unspec;  stave_size[i]:=unspec;  end;
+  { next line seems to be spurious.  0.63a RDT }
+  { begin  nspace[i]:=unspec;  stave_size[i]:=unspec;  end; }  
   n_pages:=1; n_systems:=1;
   readStyles; old_known_styles := known_styles;
   for i:=1 to lines_in_paragraph do omit_line[i]:=false;
