Index: texlive-bin/texk/web2c/man/pdftex.man
===================================================================
--- texlive-bin.orig/texk/web2c/man/pdftex.man	2019-07-05 13:07:55.356120065 +0200
+++ texlive-bin/texk/web2c/man/pdftex.man	2019-08-01 14:40:56.094286543 +0200
@@ -301,6 +301,11 @@
 .I PDF
 mode.
 .TP
+.BI -synctex \ NUMBER
+generate SyncTeX data for previewers according to bits of NUMBER. See the
+.I synctex
+manual page for details.
+.TP
 .BI -translate-file \ tcxname
 Use the
 .I tcxname
@@ -316,6 +321,9 @@
 .TP
 .B -version
 Print version information and exit.
+.TP
+.B -8bit
+make all characters printable by default.
 .\"=====================================================================
 .SH ENVIRONMENT
 See the Kpathsea library documentation (e.g., the `Path specifications'
