---
 texk/xdvik/util.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: texlive-bin-2011.20120408/texk/xdvik/util.c
===================================================================
--- texlive-bin-2011.20120408.orig/texk/xdvik/util.c	2012-04-08 21:41:48.345583375 +0900
+++ texlive-bin-2011.20120408/texk/xdvik/util.c	2012-04-08 21:42:03.009666386 +0900
@@ -259,7 +259,7 @@
     int fd = open(fname, flags);
     if (fd < 0 && (errno == EMFILE || errno == ENFILE)) {
 	close_a_file();
-	fd = open(fname, flags);
+	fd = open(fname, flags, mode);
     }
     return fd;
 }
