diff -ur xetexdir/XeTeX_ext.c xetexdir-pplib/XeTeX_ext.c
--- xetexdir/XeTeX_ext.c	Fri Jun 01 19:54:32 2018
+++ xetexdir-pplib/XeTeX_ext.c	Sun Dec 02 12:22:04 2018
@@ -38,7 +38,7 @@
 
 #include <w2c/config.h>
 
-#include <poppler-config.h>
+#include <ppapi.h>
 #include <png.h>
 #include <zlib.h>
 #include <graphite2/Font.h>
@@ -167,7 +167,7 @@
         "Compiled with Graphite2 version %d.%d.%d; using %d.%d.%d\n"
         "Compiled with HarfBuzz version %s; using %s\n"
         "Compiled with libpng version %s; using %s\n"
-        "Compiled with poppler version %s\n"
+        "Compiled with pplib version %s\n"
 #ifdef XETEX_MAC
         "Using Mac OS X Core Text and Cocoa frameworks\n"
 #else
@@ -184,7 +184,7 @@
             + strlen(hb_version_string())
             + strlen(PNG_LIBPNG_VER_STRING)
             + strlen(png_libpng_ver)
-            + strlen(POPPLER_VERSION)
+            + strlen(pplib_version)
 #ifndef XETEX_MAC
             + 6 * 3 /* for fontconfig version #s (won't really need 3 digits per field!) */
 #endif
@@ -214,7 +214,7 @@
         GR2_VERSION_MAJOR, GR2_VERSION_MINOR, GR2_VERSION_BUGFIX,
         grMajor, grMinor, grBugfix,
         HB_VERSION_STRING, hb_version_string(),
-        PNG_LIBPNG_VER_STRING, png_libpng_ver, POPPLER_VERSION
+        PNG_LIBPNG_VER_STRING, png_libpng_ver, pplib_version
 #ifndef XETEX_MAC
         ,
         FC_VERSION / 10000, (FC_VERSION % 10000) / 100, FC_VERSION % 100,

diff -ur xetexdir/pdfimage.cpp xetexdir-pplib/pdfimage.cpp
--- xetexdir/pdfimage.cpp	Tue Dec 25 14:46:34 2018
+++ xetexdir-pplib/pdfimage.cpp	Tue Dec 25 14:50:52 2018
@@ -36,9 +36,9 @@
 
 #include "pdfimage.h"
 
-#include "PDFDoc.h"
-#include "Catalog.h"
-#include "Page.h"
+extern "C" {
+#include "ppapi.h"
+}
 
 #include "XeTeX_ext.h"
 
@@ -53,22 +53,14 @@
 pdf_get_rect(char* filename, int page_num, int pdf_box, realrect* box)
 	/* return the box converted to TeX points */
 {
-	GooString*	name = new GooString(filename);
-	PDFDoc*		doc = new PDFDoc(name);
+	ppdoc*	doc = ppdoc_load(filename);
 
 	if (!doc) {
-		delete name;
 		return -1;
 	}
 
-	/* if the doc got created, it now owns name, so we mustn't delete it! */
-
-	if (!doc->isOk()) {
-		delete doc;
-		return -1;
-	}
+	int	pages = ppdoc_page_count(doc);
 
-	int	pages = doc->getNumPages();
 	if (page_num > pages)
 		page_num = pages;
 	if (page_num < 0)
@@ -76,43 +68,76 @@
 	if (page_num < 1)
 		page_num = 1;
 
-	Page*		page = doc->getCatalog()->getPage(page_num);
+	ppref*	page = ppdoc_page(doc, page_num);
+	ppdict*	pdict = ppref_obj(page)->dict;
+
+	pprect	Rect;
+	pprect*	r;
 
-	const PDFRectangle*	r;
 	switch (pdf_box) {
 		default:
 		case pdfbox_crop:
-			r = page->getCropBox();
+			r = ppdict_get_box(pdict, "CropBox", &Rect);
 			break;
 		case pdfbox_media:
-			r = page->getMediaBox();
+			r = ppdict_get_box(pdict, "MediaBox", &Rect);
 			break;
 		case pdfbox_bleed:
-			r = page->getBleedBox();
+			r = ppdict_get_box(pdict, "BleedBox", &Rect);
 			break;
 		case pdfbox_trim:
-			r = page->getTrimBox();
+			r = ppdict_get_box(pdict, "TrimBox", &Rect);
 			break;
 		case pdfbox_art:
-			r = page->getArtBox();
+			r = ppdict_get_box(pdict, "ArtBox", &Rect);
 			break;
 	}
 
-	int RotAngle = 0;
-	RotAngle = (int)page->getRotate() % 360;
+/*
+  In pplib, r can be NULL. If r == NULL, we try "CropBox",
+  "MediaBox",  "BleedBox", "TrimBox", "ArtBox" in this order.
+*/
+	if (!r) {
+		r = ppdict_get_box(pdict, "CropBox", &Rect);
+	}
+	if (!r) {
+		r = ppdict_get_box(pdict, "MediaBox", &Rect);
+	}
+	if (!r) {
+		r = ppdict_get_box(pdict, "BleedBox", &Rect);
+	}
+	if (!r) {
+		r = ppdict_get_box(pdict, "TrimBox", &Rect);
+	}
+	if (!r) {
+		r = ppdict_get_box(pdict, "ArtBox", &Rect);
+	}
+
+/*
+  If r == NULL, return error.
+*/
+	if (!r) {
+		return -1;
+	}
+
+	ppint RotAngle = 0;
+	(void)ppdict_get_int(pdict, "Rotate", &RotAngle);
+	RotAngle = RotAngle % 360;
 	if (RotAngle < 0)
 		RotAngle += 360;
 	if (RotAngle == 90 || RotAngle == 270) {
-		box->wd = 72.27 / 72 * fabs(r->y2 - r->y1);
-		box->ht = 72.27 / 72 * fabs(r->x2 - r->x1);
+		box->wd = 72.27 / 72 * fabs(r->ry - r->ly);
+		box->ht = 72.27 / 72 * fabs(r->rx - r->lx);
 	} else {
-		box->wd = 72.27 / 72 * fabs(r->x2 - r->x1);
-		box->ht = 72.27 / 72 * fabs(r->y2 - r->y1);
+		box->wd = 72.27 / 72 * fabs(r->rx - r->lx);
+		box->ht = 72.27 / 72 * fabs(r->ry - r->ly);
 	}
-	box->x  = 72.27 / 72 * my_fmin(r->x1, r->x2);
-	box->y  = 72.27 / 72 * my_fmin(r->y1, r->y2);
+	box->x  = 72.27 / 72 * my_fmin(r->lx, r->rx);
+	box->y  = 72.27 / 72 * my_fmin(r->ly, r->ry);
 
-	delete doc;
+	if (page) free(page);
+	if (pdict) free(pdict);
+	if (doc) ppdoc_free(doc);
 
 	return 0;
 }
@@ -120,21 +145,15 @@
 int
 pdf_count_pages(char* filename)
 {
-	int			pages = 0;
-	GooString*	name = new GooString(filename);
-	PDFDoc*		doc = new PDFDoc(name);
+	int	pages = 0;
+	ppdoc*	doc = ppdoc_load(filename);
 
 	if (!doc) {
-		delete name;
 		return 0;
 	}
 
-	/* if the doc got created, it now owns name, so we mustn't delete it! */
-
-	if (doc->isOk())
-		pages = doc->getNumPages();
-
-	delete doc;
+	pages = ppdoc_page_count(doc);
+	if (doc) ppdoc_free(doc);
 
 	return pages;
 }
