diff -u xetexdir-poppler/XeTeX_ext.c xetexdir/XeTeX_ext.c
--- xetexdir-poppler/XeTeX_ext.c	Fri Jun 01 19:54:32 2018
+++ xetexdir/XeTeX_ext.c	Mon Sep 24 00:27:04 2018
@@ -38,7 +38,12 @@
 
 #include <w2c/config.h>
 
+#ifdef POPPLER_VERSION
 #include <poppler-config.h>
+#else
+#include <xpdf/config.h>
+#endif
+
 #include <png.h>
 #include <zlib.h>
 #include <graphite2/Font.h>
@@ -167,7 +172,11 @@
         "Compiled with Graphite2 version %d.%d.%d; using %d.%d.%d\n"
         "Compiled with HarfBuzz version %s; using %s\n"
         "Compiled with libpng version %s; using %s\n"
+#ifdef POPPLER_VERSION
         "Compiled with poppler version %s\n"
+#else
+        "Compiled with xpdf version %s\n"
+#endif
 #ifdef XETEX_MAC
         "Using Mac OS X Core Text and Cocoa frameworks\n"
 #else
@@ -184,7 +193,11 @@
             + strlen(hb_version_string())
             + strlen(PNG_LIBPNG_VER_STRING)
             + strlen(png_libpng_ver)
+#ifdef POPPLER_VERSION
             + strlen(POPPLER_VERSION)
+#else
+            + strlen(xpdfVersion)
+#endif
 #ifndef XETEX_MAC
             + 6 * 3 /* for fontconfig version #s (won't really need 3 digits per field!) */
 #endif
@@ -214,7 +227,11 @@
         GR2_VERSION_MAJOR, GR2_VERSION_MINOR, GR2_VERSION_BUGFIX,
         grMajor, grMinor, grBugfix,
         HB_VERSION_STRING, hb_version_string(),
+#ifdef POPPLER_VERSION
         PNG_LIBPNG_VER_STRING, png_libpng_ver, POPPLER_VERSION
+#else
+        PNG_LIBPNG_VER_STRING, png_libpng_ver, xpdfVersion
+#endif
 #ifndef XETEX_MAC
         ,
         FC_VERSION / 10000, (FC_VERSION % 10000) / 100, FC_VERSION % 100,

diff -u xetexdir-poppler/pdfimage.cpp xetexdir/pdfimage.cpp
--- xetexdir-poppler/pdfimage.cpp	Tue May 29 14:56:26 2018
+++ xetexdir/pdfimage.cpp	Mon Sep 24 00:29:13 2018
@@ -53,7 +53,11 @@
 pdf_get_rect(char* filename, int page_num, int pdf_box, realrect* box)
 	/* return the box converted to TeX points */
 {
+#ifdef POPPLER_VERSION
 	GooString*	name = new GooString(filename);
+#else
+	GString*	name = new GString(filename);
+#endif
 	PDFDoc*		doc = new PDFDoc(name);
 
 	if (!doc) {
@@ -133,8 +137,12 @@
 int
 pdf_count_pages(char* filename)
 {
-	int			pages = 0;
+	int		pages = 0;
+#ifdef POPPLER_VERSION
 	GooString*	name = new GooString(filename);
+#else
+	GString*	name = new GString(filename);
+#endif
 	PDFDoc*		doc = new PDFDoc(name);
 
 	if (!doc) {
